package.json

```diff
@@ -6,38 +6,50 @@
     "type": "git",
     "url": "https://github.com/MetaMask/eslint-config.git"
   },
+  "type": "module",
```

Comment:
Is this safe because consumers always use this package as a devDep? Are we sure this doesn't affect downstream CJS packages?

Reply:
Looking at the [eslint v9 migrate guide](https://eslint.org/docs/latest/use/migrate-to-9.0.0), looks like we only need to worry about CJS configurations importing from our repo, and that shouldn't be a problem for long since eslint is now set up to use ESM by default?

Reply:
Consumers can add it and use it even if they use `type: commonjs`, as long as the config is a `.mjs` file. I figured we might as well update all places where we consume the libraries since we have to rewrite the majority of the configs anyway.